homeassistant/components/lametric/notify.py

```diff
@@ -3,157 +3,73 @@
 
-    def send_message(self, message: str = "", **kwargs: Any) -> None:
-        """Send a message to some LaMetric device."""
+    async def async_send_message(self, message: str = "", **kwargs: Any) -> None:
+        """Send a message to a LaMetric device."""
+        if not (data := kwargs.get(ATTR_DATA)):
+            data = {}
 
-        targets = kwargs.get(ATTR_TARGET)
-        data = kwargs.get(ATTR_DATA)
-        LOGGER.debug("Targets/Data: %s/%s", targets, data)
-        icon = self._icon
-        cycles = self._cycles
-        sound = None
-        priority = self._priority
-        icon_type = self._icon_type
-
-        # Additional data?
-        if data is not None:
-            if "icon" in data:
-                icon = data["icon"]
-            if "sound" in data:
-                try:
-                    sound = Sound(category="notifications", sound_id=data["sound"])
-                    LOGGER.debug("Adding notification sound %s", data["sound"])
-                except AssertionError:
-                    LOGGER.error("Sound ID %s unknown, ignoring", data["sound"])
-            if "cycles" in data:
-                cycles = int(data["cycles"])
-            if "icon_type" in data:
-                if data["icon_type"] in AVAILABLE_ICON_TYPES:
-                    icon_type = data["icon_type"]
-                else:
-                    LOGGER.warning(
-                        "Priority %s invalid, using default %s",
-                        data["priority"],
-                        priority,
-                    )
-            if "priority" in data:
-                if data["priority"] in AVAILABLE_PRIORITIES:
-                    priority = data["priority"]
-                else:
-                    LOGGER.warning(
-                        "Priority %s invalid, using default %s",
-                        data["priority"],
-                        priority,
-                    )
-        text_frame = SimpleFrame(icon, message)
-        LOGGER.debug(
-            "Icon/Message/Cycles/Lifetime: %s, %s, %d, %d",
-            icon,
-            message,
-            self._cycles,
-            self._lifetime,
-        )
+        try:
+            sound = None
+            if CONF_SOUND in data:
+                sound = Sound(id=data[CONF_SOUND], category=None)
 
-        frames = [text_frame]
+            notification = Notification(
+                icon_type=NotificationIconType(data.get(CONF_ICON_TYPE, "none")),
+                priority=NotificationPriority(data.get(CONF_PRIORITY, "info")),
+                model=Model(
+                    frames=[
+                        Simple(
+                            icon=data.get(CONF_ICON, "a7956"),
+                            text=message,
+                        )
+                    ],
+                    cycles=int(data.get(CONF_CYCLES, 1)),
+                    sound=sound,
+                ),
+            )
+        except ValueError as ex:
```

Comment:
What will raise `ValueError`?

Reply:
The values passes to the enums

Reply:
Then it's just the user input that is wrong, and we should raise `ValueError` with a message as argument?

Reply:
Right, in that case there is nothing to catch 👍